src/Arcus.Scripting.IntegrationAccount/Scripts/Set-AzIntegrationAccountAgreements.ps1

```diff
@@ -0,0 +1,87 @@
+Param(
+    [Parameter(Mandatory = $true)][string] $ResourceGroupName = $(throw "Resource group name is required"),
+    [Parameter(Mandatory = $true)][string] $Name = $(throw "Name of the Integration Account is required"),
+    [parameter(Mandatory = $false)][string] $AgreementFilePath = $(if ($AgreementsFolder -eq '') { throw "Either the file path of a specific agreement or the file path of a folder containing multiple agreements is required, e.g.: -AgreementFilePath 'C:\Agreements\agreement.json' or -AgreementsFolder 'C:\Agreements'" }),
+    [parameter(Mandatory = $false)][string] $AgreementsFolder = $(if ($AgreementFilePath -eq '') { throw "Either the file path of a specific agreement or the file path of a folder containing multiple agreements is required, e.g.: -AgreementFilePath 'C:\Agreements\agreement.json' or -AgreementsFolder 'C:\Agreements'" }),
+    [Parameter(Mandatory = $false)][string] $ArtifactsPrefix = ''
+)
+
+if ($AgreementFilePath -ne '' -and $AgreementsFolder -ne '') {
+    throw "Either the file path of a specific agreement or the file path of a folder containing multiple agreements is required, e.g.: -AgreementFilePath 'C:\Agreements\agreement.json' or -AgreementsFolder 'C:\Agreements'"
+}
+
+function UploadAgreement {
+    param(
+        [Parameter(Mandatory = $true)][System.IO.FileInfo] $Agreement
+    )
+
+    $agreementData = Get-Content -Raw -Path $Agreement.FullName | ConvertFrom-Json
+
+    $agreementName = $agreementData.name
+    if ($agreementName -eq $null -or $agreementName -eq '') {
+        throw 'Agreement name is empty'
+    }
+
+    if ($ArtifactsPrefix -ne '') {
+        $agreementName = $ArtifactsPrefix + $agreementName
+    }
+    Write-Host "Uploading agreement '$agreementName' into the Integration Account '$Name'"
+
+    $agreementType = $agreementData.properties.agreementType
+    $hostPartner = $agreementData.properties.hostPartner
+    $hostIdentityQualifier = $agreementData.properties.hostIdentity.qualifier
+    $hostIdentityQualifierValue = $agreementData.properties.hostIdentity.value
+    $guestPartner = $agreementData.properties.guestPartner    
+    $guestIdentityQualifier = $agreementData.properties.guestIdentity.qualifier
+    $guestIdentityQualifierValue = $agreementData.properties.guestIdentity.value
+    $agreementContent = $agreementData.properties.content | ConvertTo-Json -Depth 20 -Compress
+
+    $existingAgreement = $null
+    try {
+        Write-Verbose "Checking if the agreement '$agreementName' already exists in the Integration Account '$Name'"
```

Comment:
```suggestion
        Write-Verbose "Checking if the agreement '$agreementName' already exists in the Azure Integration Account '$Name'"
```

Reply:
I changed the throw message, also had to change it for the Partner script as well since that one used the same kind of message